crates/roc_host/src/roc.rs

```diff
@@ -945,3 +886,89 @@ pub extern "C" fn roc_fx_tempDir() -> RocList<u8> {
 
     RocList::from(path_os_string_bytes.as_slice())
 }
+
+pub struct RocServer {
+    pub model: roc_std::SendSafeRocList<u8>,
+    pub captures: roc_std::SendSafeRocList<u8>,
+}
+
+unsafe impl Sync for RocServer {}
+
+pub fn call_roc_init() -> RocServer {
+    extern "C" {
+        fn roc__forHost_1_exposed_generic(_: *mut u8);
+        fn roc__forHost_1_exposed_size() -> usize;
+        fn roc__forHost_0_caller(flags: *const (), captures: *mut u8, model: *mut u8);
+        fn roc__forHost_0_size() -> usize;
+    }
+
+    unsafe {
+        let captures_size = roc__forHost_1_exposed_size();
+        let mut captures = RocList::from_raw_parts(
+            roc_alloc(captures_size, 0) as *mut u8,
+            captures_size,
+            captures_size,
+        );
+
+        let model_size = roc__forHost_0_size();
+        let mut model =
+            RocList::from_raw_parts(roc_alloc(model_size, 0) as *mut u8, model_size, model_size);
```

Comment:
I'm kinda amazed this works. `model` is not a list, it is a pointer. That said, I'm not sure it is correct, would need to think about this more. Given we box the model, the output size should always be the size of a single pointer.